patches/systemd-utils/0001-Add-managarm-support.patch

```diff
@@ -0,0 +1,906 @@
+ src/basic/missing_syscall.h                   |  48 ++-
+ src/basic/mountpoint-util.c                   |   4 +
+ src/basic/process-util.c                      |   4 +-
+ src/basic/raw-clone.h                         |   4 +-
+ src/basic/raw-reboot.h                        |   6 +-
+ src/basic/socket-util.h                       |   2 +-
+ src/core/exec-invoke.c                        |   1 +
+ src/libsystemd/sd-event/sd-event.c            |   2 +-
+ src/mountfsd/mountwork.c                      |   2 +
+ src/network/netdev/bridge.c                   |   2 +-
+ src/network/netdev/fou-tunnel.c               |   2 +-
+ src/network/netdev/geneve.c                   |   2 +-
+ src/network/netdev/veth.c                     |   2 +-
+ src/network/netdev/vrf.c                      |   2 +-
+ src/network/netdev/wireguard.c                |   2 +-
+ src/network/networkd-bridge-mdb.c             |   1 +
+ src/network/networkd-dhcp-prefix-delegation.c |   2 +
+ src/network/networkd-route.c                  |   2 +
+ src/nsresourced/userns-registry.c             |   2 +
+ src/shared/bpf-program.h                      |   2 +-
+ src/shared/dev-setup.c                        |   1 +
+ src/test/test-path-util.c                     |   2 +-
+ src/udev/udevadm-settle.c                     |   2 +-
+ 25 files changed, 119 insertions(+), 284 deletions(-)
+
+diff --git a/meson.build b/meson.build
+index 5a5ac35..6883cd6 100644
+--- a/meson.build
++++ b/meson.build
+@@ -367,21 +367,21 @@ possible_common_cc_flags = [
+         '-Warray-bounds=2',
+         '-Wdate-time',
+         '-Wendif-labels',
+-        '-Werror=format=2',
+-        '-Werror=format-signedness',
+-        '-Werror=implicit-function-declaration',
+-        '-Werror=implicit-int',
+-        '-Werror=incompatible-pointer-types',
+-        '-Werror=int-conversion',
+-        '-Werror=missing-declarations',
+-        '-Werror=missing-prototypes',
+-        '-Werror=overflow',
+-        '-Werror=override-init',
+-        '-Werror=return-type',
+-        '-Werror=shift-count-overflow',
+-        '-Werror=shift-overflow=2',
+-        '-Werror=strict-flex-arrays',
+-        '-Werror=undef',
++        # '-Werror=format=2',
++        # '-Werror=format-signedness',
++        # '-Werror=implicit-function-declaration',
++        # '-Werror=implicit-int',
++        # '-Werror=incompatible-pointer-types',
++        # '-Werror=int-conversion',
++        # '-Werror=missing-declarations',
++        # '-Werror=missing-prototypes',
++        # '-Werror=overflow',
++        # '-Werror=override-init',
++        # '-Werror=return-type',
++        # '-Werror=shift-count-overflow',
++        # '-Werror=shift-overflow=2',
++        # '-Werror=strict-flex-arrays',
++        # '-Werror=undef',
+         '-Wfloat-equal',
+         # gperf prevents us from enabling this because it does not emit fallthrough
+         # attribute with clang.
+diff --git a/src/basic/linux/libc-compat.h b/src/basic/linux/libc-compat.h
```

Comment:
Instead of patching this file, we should fix mlibc to properly respect `__UAPIC_*` definitions.